doc/s3ip_sysfs/s3ip_sysfs_specification.md

```diff
@@ -0,0 +1,232 @@
+| Value  | Description                                       |
+|--------|---------------------------------------------------|
+| 0      | dark                                              |
+| 1      | green                                             |
+| 2      | yellow                                            |
+| 3      | red                                               |
+| 4      | blue                                              |
+| 5      | green light flashing                              |
+| 6      | yellow light flashing                             |
+| 7      | red light flashing                                |
+| 8      | blue light flashing                               |
+
+### 2. Temperature sensor sysfs
+The Sysfs path for temperature information must be /sys_switch/temp_sensor/
+
+*Table2-1 Temperature Sensor SYSFS property*
+
+|Sysfs path|Permissions|Data type|Description|
+|-|-|-|-|
+/sys_switch/temp_sensor/number| RO| int| equipment the total number of temperature sensors
+/sys_switch/temp_sensor/temp[n]/alias| RO| string| identify temperature point location
+/sys_switch/temp_sensor/temp[n]/type| RO| string| temperature sensor model
+/sys_switch/temp_sensor/temp[n]/max| R/W| int| alarm threshold, unit: millidegree Celsius
+/sys_switch/temp_sensor/temp[n]/min| R/W| int| alarm threshold, unit: millidegree Celsius
+/sys_switch/temp_sensor/temp[n]/value| RO| int| current temperature, unit: millidegree Celsius
+
+### 3. Voltage sensor sysfs
+Voltage Sensor information Sysfs path must be /sys_switch/vol_sensor/
+
+*Table3-1 Voltage Sensor SYSFS property *
+
+|Sysfs path|Permissions|Data type|Description|
+|-|-|-|-|
+/sys_switch/vol_sensor/number|RO|int|Total number of voltage sensors on the device
+/sys_switch/vol_sensor/vol[n]/alias|RO|string|Identifying the location of the voltage sensor
+/sys_switch/vol_sensor/vol[n]/type|RO|string|Model of the voltage sensor
+/sys_switch/vol_sensor/vol[n]/max|R/W|int|Alarm threshold, unit: mV
+/sys_switch/vol_sensor/vol[n]/min|R/W|int|Alarm recovery threshold, unit: mV
+/sys_switch/vol_sensor/vol[n]/range|RO|int|Power output error value, such as +-500mV
+/sys_switch/vol_sensor/vol[n]/nominal_value|RO|int|Nominal value of supply voltage, unit: mV
+/sys_switch/vol_sensor/vol[n]/value|RO|int|Voltage value, unit: mV
+
+### 4. Current sensor sysfs
+Current Sensor information Sysfs path must be /sys_switch/curr_sensor/
+
+*Table4-1 Current Sensor SYSFS property*
+
+|Sysfs path|Permissions|Data type|Description|
+|-|-|-|-|
+/sys_switch/curr_sensor/number|RO|int|equipment is the total number of current sensor
+/sys_switch/curr_sensor/curr[n]/alias|RO|string|Identifying the location of the current sensor
+/sys_switch/curr_sensor/curr[n]/type|RO|string|current sensor model
+/sys_switch/curr_sensor/curr[n]/max|R/W|int|Alarm threshold, unit: mA
+/sys_switch/curr_sensor/curr[n]/min|R/W|int|Alarm recovery threshold, unit: mA
+/sys_switch/curr_sensor/curr[n]/value|RO|int|current unit: mA
+

```

Comment:
I suggest adding a `Power Sensor sysfs` section. This would provide the power in milliwatts from various power-sensing devices throughout the platform. Most on-board DC-DC converters (also called PWM or pulse-width modulator chips) provide voltage, current and calculated power. They often provide power In and Power out, allowing easy measurement of efficiency. It might be more reliable to read this directly rather than compute it from current * voltage, which might obtain readings at different times and result in invalid computation. It could also be more accurate because multiplying current * power in the controller results in compound loss of precision (rounded value * rounded value).

Reply:
Prior to this, the requirement to directly collect the Power sensor was not collected (because it can be obtained normally via Current *voltage), so it was not covered in the first phase.

This is a good point, we can probably add this sysfs definition in the next stage.